src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,15 +37,29 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] splitup = trimmedAddress.split(",");
+        createClasses(splitup);
```

Comment:
You attempted to apply SLAP.  However, realize that despite you wanted to make it high level, it's not consistent with the rest of the code anymore.  Remember, the objective of SLAP is not only high level, but also same level.